### ISSUE TITLE
[FEATURE REQ] Automatically run Export-API.ps1 in Pull Requests

### ISSUE THREAD

**Issue by kasobol-msft** [Azure/azure-sdk-for-net#10997]
Would it be possible to automagically run Export-Api.ps1 script in Pull Requests and in CI pipelines?
It's quite annoying to do this manually. If one forgets to do that then it's waste of time and unnecessary PR iteration.

**Comment by tg-msft** [comment]
That would require the CI pipeline to checkout files and make new commits.  We don't have any infrastructure like that in place and I'd be hesitant to add any.

As a workaround, you can add a git hook that verifies Export-Api, Update-Snippets, etc., have been called for the specific project you care about.  You can export APIs for individual projects with something like:

```ps
pushd C:\src\sdks\net\sdk\search\Azure.Search.Documents\src
dotnet build /p:GenerateApiListingOnBuild=true /p:Configuration=Release
popd
```

and you can update snippets for a subtree with something like:

```ps
pushd C:\src\sdks\net\eng
./Update-Snippets.ps1 search
popd
```

**Comment by tg-msft** [comment]
I'm actually going to close this issue as "by design."  We should never be doing things that make it easy to ignore API changes.

I've filed https://github.com/Azure/azure-sdk-for-net/issues/10998 to add similar quality of life improvements in a safer way.

**Comment by kasobol-msft** [comment]
@tg-msft  thanks for workaround suggestions. I'll try them out.
The intention here was like #10998 you just created - to seek automation end to end for API files. We don't edit those files manually - they're more "build artifacts" rather than "sources", so it should be possible to invent some process that doesn't involve human interaction (but still provides diffs and awareness of changes happening).

**Comment by chidozieononiwu** [comment]
Please see updated contributing docs for how to UpdateSourceOnBuild https://github.com/Azure/azure-sdk-for-net/blob/master/CONTRIBUTING.md#updating-source-on-build

**Comment by kasobol-msft** [comment]
great. thank you!

**Comment by kasobol-msft** [comment]
@chidozieononiwu 
So I tried that with https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/storage/Azure.Storage.sln and it isn't working well.

1. There are some errors from Azure.Core (?)
2. It seems that somehow the script got into infinite recursion.

See screenshots:

![image](https://user-images.githubusercontent.com/61715331/89805238-35522600-daea-11ea-882a-345c893c2cce.png)

![image](https://user-images.githubusercontent.com/61715331/89805273-41d67e80-daea-11ea-85da-1d2c873651d4.png)

**Comment by kurtzeborn** [comment]
@chidozieononiwu, looks like this may be something not quite right with the fix for #10998.
